modin/pandas/dataframe.py

```diff
@@ -160,7 +162,18 @@ def __init__(
                 if columns is None:
                     columns = slice(None)
                 self._query_compiler = data.loc[index, columns]._query_compiler
-
+        elif isinstance(data, array):
+            self._query_compiler = data._query_compiler.copy()
+            if copy is not None and not copy:
+                data._add_sibling(self)
+            if columns is not None and not isinstance(columns, pandas.Index):
+                columns = pandas.Index(columns)
+            if columns is not None:
+                self.set_axis(columns, axis=1, inplace=True)
+            if index is not None:
+                self.set_axis(index, axis=0, inplace=True)
+            if dtype is not None:
+                self.astype(dtype, copy=False)
```

Comment:
shouldn't we either pass `inplace=True` or assign the result somewhere?

Reply:
Setting `copy=False` makes the change happen inplace! (We do test that this works, see the test where we pass dtype=str).